pandas/_libs/tslib.pyx

```diff
@@ -314,8 +330,33 @@ def array_with_unit_to_datetime(
                     iresult[i] = NPY_NAT
 
                 else:
+
+                    try:
+                        fval = float(val)
+                    except ValueError:
+                        if is_raise:
+                            raise ValueError(
+                                f"non convertible value {val} with the unit '{unit}'"
+                            )
+                        elif is_ignore:
+                            raise AssertionError
+                        iresult[i] = NPY_NAT
+                        continue
+
+                    if is_ym and not fval.is_integer():
+                        # Analogous to GH#47266 for Timestamp
+                        if is_raise:
+                            raise ValueError(
+                                f"Conversion of non-round float with unit={unit} "
+                                "is ambiguous"
+                            )
+                        elif is_ignore:
+                            raise AssertionError
+                        iresult[i] = NPY_NAT
+                        continue
```

Comment:
sorry to block this, but is there a test that hits this?

I can see tests when `val` is a non-round float, but what about when it's a string containing a non-round float (e.g. `'1.5'`), which I believe is what would reach this branch?